types/src/contracts/entry_point_type.rs

```diff
@@ -0,0 +1,71 @@
+#![allow(non_upper_case_globals)] // bitflags 2.x causes wasm bloat, so we need this attribute
+
+use alloc::vec::Vec;
+
+#[cfg(feature = "datasize")]
+use datasize::DataSize;
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+#[cfg(feature = "json-schema")]
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::bytesrepr::{self, FromBytes, ToBytes};
+
+/// Context of method execution
+///
+/// First bit represents version i.e.
```

Comment:
Isn't it the "last" bit, technically? :)
Maybe consider referring to it as "most significant bit"?